cpp/test/sg/rf_treelite_test.cu

```diff
@@ -0,0 +1,397 @@
+
+    TREELITE_CHECK(TreeliteDeleteDenseBatch(dense_batch));
+    TREELITE_CHECK(TreelitePredictorFree(predictor));
+    TREELITE_CHECK(TreeliteFreeModel(model));
+  }
+
+  void getResultAndCheck() {
+    // Predict and compare against known labels
+    RF_metrics tmp =
+      score(handle, forest, inference_data_d, labels_d, params.n_inference_rows,
+            params.n_cols, predicted_labels_d, false);
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+
+    if (is_classification) {
+      predicted_labels_h.resize(params.n_inference_rows);
+      CUDA_CHECK(cudaMemcpy(predicted_labels_h.data(), predicted_labels_d,
+                            sizeof(L) * params.n_inference_rows,
+                            cudaMemcpyDeviceToHost));
+    } else {
+      // In case of regression, predicted_labels_d alreay has the final result.
+      ref_predicted_labels.resize(params.n_inference_rows);
+      CUDA_CHECK(cudaMemcpy(ref_predicted_labels.data(), predicted_labels_d,
+                            sizeof(L) * params.n_inference_rows,
+                            cudaMemcpyDeviceToHost));
+    }
+
+    if (is_classification) {
+      for (int i = 0; i < params.n_inference_rows; i++) {
+        ref_predicted_labels[i] = static_cast<float>(predicted_labels_h[i]);
+        treelite_predicted_labels[i] =
+          treelite_predicted_labels[i] >= 0.5 ? 1 : 0;
+      }
+    }
+
+    EXPECT_TRUE(devArrMatchHost(
+      ref_predicted_labels.data(), treelite_predicted_labels.data(),
+      params.n_inference_rows, Compare<float>(), stream));
+  }
+
+  void SetUp() override {
+    params = ::testing::TestWithParam<RfInputs<T>>::GetParam();
+
+    DecisionTree::DecisionTreeParams tree_params;
+    set_tree_params(tree_params, params.max_depth, params.max_leaves,
+                    params.max_features, params.n_bins, params.split_algo,
+                    params.min_rows_per_node, params.bootstrap_features,
+                    params.split_criterion, false);
+    set_all_rf_params(rf_params, params.n_trees, params.bootstrap,
+                      params.rows_sample, tree_params);
+    // print(rf_params);
+
+    data_len = params.n_rows * params.n_cols;
+    inference_data_len = params.n_inference_rows * params.n_cols;
+
+    allocate(data_d, data_len);
+    allocate(inference_data_d, inference_data_len);
+
+    allocate(labels_d, params.n_rows);
+    allocate(predicted_labels_d, params.n_inference_rows);
+
+    treelite_predicted_labels.resize(params.n_inference_rows);
+    ref_predicted_labels.resize(params.n_inference_rows);
+
+    CUDA_CHECK(cudaStreamCreate(&stream));
+    handle.setStream(stream);
+
+    forest = new typename ML::RandomForestMetaData<T, L>;
+    null_trees_ptr(forest);
+
+    data_h.resize(data_len);
+    inference_data_h.resize(inference_data_len);
+
+    for (int i = 0; i < params.n_rows; i++) {
+      for (int j = 0; j < params.n_cols; j++) {
+        float base = j * 100.0;
```

Comment:
It's better to just generate the data randomly (uniform or normal), but use the labels that are a linear function of their data row + noise.